src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -37,6 +60,43 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
                 .map(s -> SortOrder.valueOf(sortOrder.toUpperCase()))
                 .orElse(null);
         request.setAttribute(PRODUCTS, productDao.findProducts(query, field, order));
+        List<Product> productList = recentViewed
+                .getRecentViewed(httpSession)
+                .getItems();
+        request.setAttribute(RECENT_VIEWED, productList
+                .stream()
+                .filter(p -> {
+                    if (productList.size() < 4) return true;
```

Comment:
Use full version of if/for/while operators

Reply:
But why can't I use streem?

Reply:
if(condition) {
    statement();
}